### PR TITLE
[Core] Integration of Rizin SigDB in aaa and add aaF

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- Adds `aaF` and `aaFl` to manually apply the signatures.
- Adds automatic application of the signatures via the sigdb repository when `flirt.sigdb.path` is set.

image

Added test

### CODE REVIEW

File: librz/core/canalysis.c [hunk 1]
```diff
@@ -5940,6 +5940,119 @@ RZ_API bool rz_core_analysis_everything(RzCore *core, bool experimental, char *d
...
+	return true;
+}
+
+/**
+ * \brief Outputs the list of signatures found in the flirt.sigdb.path
+ *
+ * \param core The RzCore instance
+ */
+RZ_API void rz_core_analysis_sigdb_list(RzCore *core) {
+	const char *sigdb_path = rz_config_get(core->config, "flirt.sigdb.path");
```
Comment (XVilka) [inline librz/core/canalysis.c]: Probably better to separate loading signatures and listing them. Like we do with type databases, for example.

**Comment by thestr4ng3r** [comment]
> Regarding testing, i'm not sure how to do this, but i guess this is ok.

You can put a small stub sigdb into the testbins and write an rz-test test that sets the path into the testbins, runs `aaa` on a bin and checks if everything matched right.

File: librz/core/canalysis.c (hunk 1 above)
Comment (ret2libc) [reply librz/core/canalysis.c]: If you mean having a `RzList<RzAnalysisSignature> *rz_analysis_sigdb_list` and then a `_print` method that prints that list of structures, then I agree.

**Review by thestr4ng3r — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
```
[XX] C:\projects\rizin\test\db\cmd\cmd_aaF Print SigDB contents
ANSICON=1 RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'e flirt.sigdb.path=bins/sigdb/fake-db
aaFl
' =
-- stdout
--- expected
+++ actual
@@ -1,4 +1,4 @@
-bin arch bits name          
-----------------------------
-elf x86  64   not-match.pat
-elf x86  64   match.sig
+bin arch bits            name                             
+----------------------------------------------------------
+elf x86  140694538682432 fake-db\elf\x86\64\match.sig
+elf x86  140694538682432 fake-db\elf\x86\64\not-match.pat
[**]               C:\projects\rizin\test\db\cmd\cmd_ab     4239 OK       123 BR        0 XX       15 FX
[XX] C:\projects\rizin\test\db\cmd\cmd_aaF Do analysis and apply SigDB manually
ANSICON=1 RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'aaa
e flirt.sigdb.path=bins/sigdb/fake-db
aaF match.sig
afl~printf
' bins/sigdb/elf-x86-64-static-libc
-- stdout
--- expected
+++ actual
@@ -1,3 +1,3 @@
-Applying elf/x86/64/match.sig signature file
+Applying elf\x86\64\match.sig signature file
 0x00409c70   18 516  -> 514  flirt.printf
 0x0046fc50    7 516  -> 342  flirt.printf
-- stderr
[ ] Analyze all flags starting with sym. and entry0 (aa)[[x] Analyze all flags starting with sym. and entry0 (aa)
[ ] Analyze function calls (aac)[[x] Analyze function calls (aac)
[ ] Analyze len bytes of instructions for references (aar)[[x] Analyze len bytes of instructions for references (aar)
[ ] Check for classes[[x] Check for classes
[ ] Type matching analysis for all functions (aaft)[[x] Type matching analysis for all functions (aaft)
[ ] Propagate noreturn information[[x] Propagate noreturn information
[ ] Use -AA or aaaa to perform additional experimental analysis.[[x] Use -AA or aaaa to perform additional experimental analysis.
[ ] Applying signatures from sigdb[[x] Applied 0 FLIRT signatures via sigdb
```
https://ci.appveyor.com/project/rizinorg/rizin/builds/41952792/job/ld0l6pbd7p788nbx#L4278

**Pull request merged**